src/ode_solver/relaxation_runge_kutta/runge_kutta_store_entropy.cpp

```diff
@@ -0,0 +1,176 @@
+#include "rrk_ode_solver_base.h"
+#include "physics/euler.h"
+#include "physics/physics_factory.h"
+
+namespace PHiLiP {
+namespace ODE {
+
+template <int dim, typename real, typename MeshType>
+RKNumEntropy<dim,real,MeshType>::RKNumEntropy(
+            std::shared_ptr<RKTableauBase<dim,real,MeshType>> rk_tableau_input)
+        : EmptyRRKBase<dim,real,MeshType>(rk_tableau_input)
+        , butcher_tableau(rk_tableau_input)
+        , n_rk_stages(butcher_tableau->n_rk_stages)
+        , mpi_communicator(MPI_COMM_WORLD)
+        , mpi_rank(dealii::Utilities::MPI::this_mpi_process(MPI_COMM_WORLD))
+        , pcout(std::cout, mpi_rank==0)
+{
+    this->rk_stage_solution.resize(n_rk_stages);
+   
+}
+
+template <int dim, typename real, typename MeshType>
+void RKNumEntropy<dim,real,MeshType>::store_stage_solutions(const int istage, const dealii::LinearAlgebra::distributed::Vector<double> rk_stage_i)
+{
+    //Store the solution value
+    //This function is called before rk_stage is modified to hold the time-derivative
+    this->rk_stage_solution[istage] = rk_stage_i; 
+}
+
+template <int dim, typename real, typename MeshType>
+dealii::LinearAlgebra::distributed::Vector<double> RKNumEntropy<dim,real,MeshType>::compute_entropy_vars(const dealii::LinearAlgebra::distributed::Vector<double> &u,
+        std::shared_ptr<DGBase<dim,real,MeshType>> dg) const
+{
+    // hard-code nstate for Euler/NS - ODESolverFactory has already ensured that we use Euler/NS
+    const unsigned int nstate = dim + 2;
+    // Currently only implemented for constant p
+    const unsigned int poly_degree = dg->get_max_fe_degree();
+    if (poly_degree != dg->get_min_fe_degree()){
+        this->pcout << "Error: Entropy RRK is only implemented for uniform p. Aborting..." << std::endl;
+        std::abort();
+    }
+    
+    // Select Euler physics. ODESolverFactory has already ensured that we are using Euler orSelect Euler physics. ODESolverFactory has already ensured that we are using Euler or NS, which both use the same entropy variable computation.
+    PHiLiP::Parameters::AllParameters parameters_euler = *(dg->all_parameters);
```

Comment:
Changed to `parameters_using_euler_pde_type`. I don't want to use `all_parameters` to avoid confusion considering we use that name elsewhere, and this parameters object is only used to declare a physics class!